spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java

```diff
@@ -1479,6 +1483,10 @@ public void setProducerConfigCustomizer(ProducerConfigCustomizer producerConfigC
 		this.producerConfigCustomizer = producerConfigCustomizer;
 	}
 
+	public List<AbstractMessageListenerContainer<?, ?>> getKafkaMessageListenerContainers() {
+		return kafkaMessageListenerContainers;
```

Comment:
We should return `Collections.unModifiableList()` here.

It should also be package private - we don't want end users using this API.

Reply:
done